### PR TITLE
fix: Disallow delegations in the past.

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

File: libs/dh/market-participant/actors/feature-delegation/src/lib/create/dh-delegation-create-modal.component.html [hunk 1]
```diff
@@ -43,7 +43,16 @@
       <watt-datepicker-v2
         [label]="t('start')"
         [formControl]="createDelegationForm.controls.startDate"
-      />
+        [min]="date"
+      >
+        <watt-field-error
+          *ngIf="
```
Comment (dzhavat) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/create/dh-delegation-create-modal.component.html:49]: suggestion: let's use the new `@if` syntax

File: libs/dh/market-participant/actors/feature-delegation/src/lib/stop/dh-delegation-stop-modal.component.ts [hunk 2]
```diff
@@ -100,7 +105,15 @@ import { DhDelegation } from '../dh-delegations';
           >
             {{ t('stopDate') }}
           </watt-radio>
-          <watt-datepicker-v2 [min]="date" [formControl]="stopDelegationForm.controls.stopDate" />
+          <watt-datepicker-v2 [min]="date" [formControl]="stopDelegationForm.controls.stopDate">
+            <watt-field-error
+              *ngIf="
```
Comment (dzhavat) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/stop/dh-delegation-stop-modal.component.ts:110]: suggestion: let's use the new `@if` syntax

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-validators.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export const dateCannotBeOlderThanTodayValidator =
+  () =>
+  (control: AbstractControl): ValidationErrors | null => {
+    const controlStart = control.value as string;
+    if (!controlStart) return null;
+
+    const startDate = dayjs(controlStart).toDate();
+
+    if (startDate < dayjs().startOf('day').toDate()) {
```
Comment (dzhavat) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-validators.ts:28]: suggestion: I think DayJS has a method to compare two dates. Let's use that instead.

Comment (defectiveAi) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-validators.ts]: fixed

File: libs/dh/market-participant/actors/feature-delegation/src/lib/create/dh-delegation-create-modal.component.html (hunk 1 above)
Comment (defectiveAi) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/create/dh-delegation-create-modal.component.html]: fixed

**Review by mimse — APPROVED** [review]

**Pull request merged**